lib/application.js

```diff
@@ -423,7 +423,7 @@ function dataSourcesFromConfig(name, config, connectorRegistry, registry) {
         config.connector = require(connectorPath);
       }
     }
-    if (!config.connector.name)
+    if (typeof config.connector !== 'string' && !config.connector.name)
```

Comment:
I was also thinking of using:
`    if (!config.connector)` 
^ Did you try that?